x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -376,6 +393,21 @@ private void executeSubPlans(
                 // Ensure we don't have subplan flag stuck in there on failure
                 ActionListener.runAfter(listener, executionInfo::finishSubPlans)
             );
+        } else if (statement.setting(QuerySettings.APPROXIMATE)) {
+            new Approximate(
+                optimizedPlan,
+                logicalPlanOptimizer,
+                p -> logicalPlanToPhysicalPlan(
+                    optimizedPlan(p, logicalPlanOptimizer, planTimeProfile),
```

Comment:
We're breaking away from the standard flow quite a bit here. I'm concerned that this branch has access to the optimizer. That looks... wrong. We should already be done optimizing here, and not pass down an optimizer into the execute methods.

Why is this needed? I didn't fully grok the Approximate class, yet, but unless it's is really complicated, we should find a way to not require approximated queries to have their own flow.